operate/schema/src/main/java/io/camunda/operate/schema/opensearch/OpensearchSchemaManager.java

```diff
@@ -190,7 +201,90 @@ public String getIndexPrefix() {
     return operateProperties.getOpensearch().getIndexPrefix();
   }
 
-  private void createDefaults() {
+  @Override
+  public Map<String, IndexMapping> getIndexMappings(final String indexNamePattern) {
+    return richOpenSearchClient.index().getIndexMappingsNew(indexNamePattern);
+  }
+
+  @Override
+  public void updateSchema(final Map<IndexDescriptor, Set<IndexMappingProperty>> newFields) {
+    for (Map.Entry<IndexDescriptor, Set<IndexMappingProperty>> indexNewFields :
+        newFields.entrySet()) {
+      if (indexNewFields.getKey() instanceof TemplateDescriptor) {
+        LOGGER.info(
+            "Update template: " + ((TemplateDescriptor) indexNewFields.getKey()).getTemplateName());
+        final TemplateDescriptor templateDescriptor = (TemplateDescriptor) indexNewFields.getKey();
+        final String json = readTemplateJson(templateDescriptor);
+        final PutIndexTemplateRequest indexTemplateRequest =
+            prepareIndexTemplateRequest(templateDescriptor, json);
+        putIndexTemplate(indexTemplateRequest);
+      }
+
+      final Map<String, Property> properties;
+      try (JsonParser jsonParser =
+          JsonProvider.provider()
+              .createParser(
+                  new StringReader(
+                      IndexMappingProperty.toJsonString(
+                          indexNewFields.getValue(), objectMapper)))) {
+        properties =
+            JsonpDeserializer.stringMapDeserializer(Property._DESERIALIZER)
+                .deserialize(jsonParser, jsonpMapper);
+      }
+      PutMappingRequest request =
+          new PutMappingRequest.Builder()
+              .index(indexNewFields.getKey().getAlias())
+              .properties(properties)
+              .build();
+      LOGGER.info(
+          String.format(
+              "Index alias: %s. New fields will be added: %s",
+              indexNewFields.getKey().getAlias(), indexNewFields.getValue()));
+      richOpenSearchClient.index().putMapping(request);
+    }
+  }
+
+  @Override
+  public IndexMapping getExpectedIndexFields(final IndexDescriptor indexDescriptor) {
+    final String schemaFilename;
+    if (indexDescriptor instanceof TemplateDescriptor) {
+      schemaFilename =
+          String.format(
+              SCHEMA_FOLDER + "/template/operate-%s.json", indexDescriptor.getIndexName());
+    } else {
+      schemaFilename =
+          String.format(SCHEMA_FOLDER + "/index/operate-%s.json", indexDescriptor.getIndexName());
+    }
+    final InputStream description =
+        ElasticsearchSchemaManager.class.getResourceAsStream(schemaFilename);
+
+    try {
+      final String currentVersionSchema =
+          StreamUtils.copyToString(description, Charset.defaultCharset());
```

Comment:
I suggest to either use the files' actual charset or just pass the `InputStream` to Jackson. Same pattern is in `ElasticsearchSchemaManager`.